# Реализуйте декоратор takes_positive, который проверяет, что все аргументы, передаваемые в декорируемую функцию,
# являются положительными целыми числами.

# Если хотя бы один аргумент не удовлетворяет данному условию, декоратор должен возбуждать исключение:

# TypeError, если аргумент не является целым числом
# ValueError, если аргумент является целым числом, но отрицательным или равным нулю
# Примечание 1. Приоритет возбуждения исключений при несоответствии аргумента обоим условиям или при наличии разных
# аргументов, несоответствующих разным условиям: TypeError, затем ValueError.

# Примечание 2. Не забывайте про то, что декоратор не должен поглощать возвращаемое значение декорируемой функции,
# а также должен уметь декорировать функции с произвольным количеством позиционных и именованных аргументов.



def takes_positive(func):
    def wrapper(*args, **kwargs):
        if any([isinstance(x, float) for x in [*args, *kwargs.values()]]):
            raise TypeError
        if any([x <= 0 for x in [*args, *kwargs.values()]]):
            raise ValueError
        return func(*args, **kwargs)
    return wrapper